### PR TITLE
[#2824] bugFix: Fixed abnormal timestamp type conversion between iceberg and gravitino

### PR DESCRIPTION
**Pull request by caican00**
### What changes were proposed in this pull request?

in com.datastrato.gravitino.catalog.lakehouse.iceberg.converter.ConvertUtil#formIcebergType

```
Types.TimestampType.withZone() -> com.datastrato.gravitino.rel.types.Types.TimestampType.withTimeZone()

Types.TimestampType.withoutZone() -> com.datastrato.gravitino.rel.types.Types.TimestampType.withoutTimeZone()
```

### Why are the changes needed?

Fixed abnormal timestamp type conversion between iceberg and gravitino

Fix: https://github.com/datastrato/gravitino/issues/2824

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Existing Unit tests and new IT.

### CODE REVIEW

**Comment by caican00** [comment]
Hi @FANNG1 could you help review this PR? fix an timestampType conversion exception.

File: catalogs/catalog-lakehouse-iceberg/src/test/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/converter/TestConvertUtil.java [hunk 1]
```diff
@@ -278,12 +278,20 @@ public void testFormIcebergType() {
     Assertions.assertTrue(
```
Comment (FANNG1) [inline catalogs/catalog-lakehouse-iceberg/src/test/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/converter/TestConvertUtil.java:278]: could you add a integrate test in `CatalogIcebergIT`?

Comment (caican00) [reply catalogs/catalog-lakehouse-iceberg/src/test/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/converter/TestConvertUtil.java:278]: updated.

File: catalogs/catalog-lakehouse-iceberg/src/test/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/integration/test/CatalogIcebergIT.java [hunk 2]
```diff
@@ -439,6 +439,61 @@ void testCreateAndLoadIcebergTable() {
...
+    Assertions.assertEquals("iceberg_column_1", loadTable.columns()[0].name());
+    Assertions.assertEquals(Types.TimestampType.withTimeZone(), loadTable.columns()[0].dataType());
+    Assertions.assertEquals("col_1_comment", loadTable.columns()[0].comment());
+
+    Assertions.assertEquals("iceberg_column_2", loadTable.columns()[1].name());
+    Assertions.assertEquals(
+        Types.TimestampType.withoutTimeZone(), loadTable.columns()[1].dataType());
+    Assertions.assertEquals("col_2_comment", loadTable.columns()[1].comment());
+
+    org.apache.iceberg.Table table =
```
Comment (FANNG1) [inline catalogs/catalog-lakehouse-iceberg/src/test/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/integration/test/CatalogIcebergIT.java:481]: why check hive catalog here?

Comment (caican00) [reply catalogs/catalog-lakehouse-iceberg/src/test/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/integration/test/CatalogIcebergIT.java:481]: i want to check that the timestamp type conversion between gravitino and iceberg is correct

Comment (FANNG1) [reply catalogs/catalog-lakehouse-iceberg/src/test/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/integration/test/CatalogIcebergIT.java:481]: ok

**Review by FANNG1 — APPROVED** [review]

**Comment by FANNG1** [comment]
@caican00 , thanks for your work!

**Pull request merged**